### PR TITLE
Allow closing pull request instead of exiting with error

### PR DESCRIPTION
**Pull request by HasanHajHasan**
Adding close PR feature as an option instead of failing it

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -19,17 +19,25 @@ export async function run(): Promise<void> {
         )
       }
       const githubToken: string = core.getInput('githubToken', {required: true})
-      const gitHubService = new GitHubService(githubToken)
+      const githubService = new GithubService(githubToken)
```
Comment (xalvarez) [inline src/main.ts:22]: Would you mind not renaming this variable?

File: src/pattern-matcher.ts [hunk 2]
```diff
@@ -17,7 +22,20 @@ export async function checkChangedFilesAgainstPattern(
...
-      core.setFailed(`There is at least one file matching the pattern ${pattern}`)
+      const closePR: boolean = core.getInput('closePR') === 'true'
+      if (closePR) {
+        const octokit = github.getOctokit(token)
+        const response = await octokit.rest.pulls.update({
+          owner,
+          repo,
+          pull_number: pullRequestNumber,
+          state: 'closed'
+        })
```
Comment (xalvarez) [inline src/pattern-matcher.ts:33]: There's a GitHubService class which is responsible for interacting with GitHub. Adding this call here mixes responsibilities. I think a cleaner code would be:

```
if (closePR) {
        await githubService.closePullRequest(owner, repo, pullRequestNumber)
      } else {
        core.setFailed(`There is at least one file matching the pattern ${pattern}`)
      }
```

Additionally, the tests for this API call are missing, which is way the PR check is failing (coverage threshold isn't reached).

File: src/pattern-matcher.ts [hunk 3]
```diff
@@ -17,7 +22,20 @@ export async function checkChangedFilesAgainstPattern(
     })
 
     if (shouldPreventFileChange) {
-      core.setFailed(`There is at least one file matching the pattern ${pattern}`)
+      const closePR: boolean = core.getInput('closePR') === 'true'
```
Comment (xalvarez) [inline src/pattern-matcher.ts:25]: All inputs are read in main.ts. For consistency I would read this there too.

**Review by xalvarez — CHANGES_REQUESTED** [review]
Hello @HasanHajHasan,

Thank you for your contribution!

I added some comments to your PR. It would be great if you could address them. If you need help with the adjustments, let me know and I can take over.

**Comment by HasanHajHasan** [comment]
Hey @xalvarez,
Is there any missing or incomplete step|(s) needed to be done ? so we can proceed with the merge

**Comment by HasanHajHasan** [comment]
Hey @xalvarez,
Is there any missing or incomplete step|(s) needed to be done ? so we can proceed with the merge

**Comment by xalvarez** [comment]
Hi @HasanHajHasan, thanks for your changes! I'm currently on vacation and I can't properly review these changes right now. I'll take care of this PR as soon as I can.

**Review by xalvarez — APPROVED** [review]
Thanks for your contribution @HasanHajHasan! I'll merge these changes and prepare a release.

**Pull request merged**